### PR TITLE
Sfl 93

### PR DESCRIPTION
**Pull request by saketPandhare-Webonise**

### CODE REVIEW

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if (indexPath.section == 0) {
+            return SEARCH_CELL_HEIGHT ///Search cell height
+        }else {
+            return FILTER_CELL_HEIGHT /// Filter cell height
+        }
+    }
+    
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift:42]: Remove extra space

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        configureFilterCountOnBarItem()
+    }
+}
+
+extension FilterVC: SearchBtnClickDelegate {
+    
+    /// Delegate Method called on Search clicked
+    /// - Parameter searchText: search text
+    func didClickOnSearchBtn(searchText: String) {
+        // print("Search text has \(searchText)")
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift:122]: remove commented code

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        filterScreenDelegate?.searchBtnClicked(selectedText: searchText)
+        popVC(animated: false)
+    }
+    
+    func didClickOnClearFilter() {
+        arraySelectedFilters.removeAll()
+        filterScreenDelegate?.clearFilterClicked()
+        tableViewFilter.reloadData()
+        configureFilterCountOnBarItem()
+        //popVC(animated: false)
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift:132]: remove commented code

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterHeaderCell.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+    
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+        
+        // Configure the view for the selected state
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/FilterHeaderCell.swift:22]: It is Todo as per comment or else?

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterTableCell.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+        viewHeaderBorder.addShadowToViewInFilter()
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/FilterTableCell.swift:29]: Same here

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterVC.swift [hunk 6]
```diff
@@ -124,4 +160,19 @@ class FilterVC: NavigationController {
...
+    /// This function is used to togle filter btn depending upon filter selected
+    /// - Parameters:
+    ///   - enable: enbale/disable value
+    ///   - bgColor: bg color
+    func toggleApplyFilterBtn(enable: Bool, bgColor: UIColor) {
+        btnApply.isEnabled = enable
+        btnApply.backgroundColor = bgColor
+    }
+    
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/FilterVC.swift:173]: remove extra space

File: SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            btnDot.isHidden = true
+        }
+    }
+    
+    /// This function is used to get description for selected filters
+    /// - Returns: string
+    func getTestForFilterLbl() -> String {
+        if (arraySelectedFilter.count > 0) {
+            if (arraySelectedFilter.count == 1) {
+               return "1 Filter"
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift:79]: Can we mention reason for hardcoding or else move to constants with reason

File: SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            }else {
+                return "\(arraySelectedFilter.count) Filters"
+            }
+        }else {
+            return "0 Filters"
+        }
+    }
+    
+    func updateSelectedFilterImage() {
+
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift:89]: Add the method defination here with documentation or else remove if not required

File: SUNFLOWR/SUNFLOWR/MindVC/MindVC.swift [hunk 9]
```diff
@@ -291,14 +297,22 @@ extension MindVC: FilterScreenDelegate {
...
-            showRightBarButton(image: "FilterFilled")
-            showFilterBadgeWithCount(count: "5") //selectedFilterCnt in testing
-        }
+    func applyBtnClick(arraySelectedFilters: [IndexPath]) {
+//        if (selectedFilterCnt != "") {
+//            showRightBarButton(image: "FilterFilled")
+//            showFilterBadgeWithCount(count: "5") //selectedFilterCnt in testing
+//        }
         ///As Of Now Reset Pagination on Apply Btn Click of Filter Screen
         restPagination()
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/MindVC/MindVC.swift:306]: is it resetPagination? & hopefully the above commented code will be removed once its finalized.

File: SUNFLOWR/SUNFLOWR/MoveVC/MoveVC.swift [hunk 10]
```diff
@@ -149,7 +173,8 @@ class MoveVC: NavigationController {
             showLeftBarButtonWithImage(image: "Fav")
             removeSubviewAddedIfAny()
             ///Remove all Fav Items from Array
-            checkForPagination(shouldResetPagination: true)
+            callAPIAsPerSelectedType()
+            //checkForPagination(shouldResetPagination: true)
```
Comment (vineet-webonise) [inline SUNFLOWR/SUNFLOWR/MoveVC/MoveVC.swift:177]: Remove comented code if not required

Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/MoveVC/MoveVC.swift:177]: Done

File: SUNFLOWR/SUNFLOWR/MindVC/MindVC.swift (hunk 9 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/MindVC/MindVC.swift:306]: yes that will be removed when mind filter listing is done

File: SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift (hunk 8 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift]: Removed

File: SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift (hunk 7 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/SearchFilterCell.swift]: Done

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterVC.swift (hunk 6 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/FilterVC.swift]: Done

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterTableCell.swift (hunk 5 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/FilterTableCell.swift]: This is the default code that get added while creating cell ,Removed

File: SUNFLOWR/SUNFLOWR/FilterVC/FilterHeaderCell.swift (hunk 4 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/FilterHeaderCell.swift]: This is the default code that get added while creating cell ,Removed

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift (hunk 3 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift]: Done

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift (hunk 2 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift]: Done

File: SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift (hunk 1 above)
Comment (saketPandhare-Webonise) [reply SUNFLOWR/SUNFLOWR/FilterVC/Filter+TableView.swift]: Done

**Pull request merged**